libbindgen/src/codegen/mod.rs

```diff
@@ -1667,7 +1666,26 @@ impl CodeGenerator for Enum {
             Some(&name)
         };
 
-        for variant in self.variants().iter() {
+        // NB: We defer the creation of constified variants, in case we find
+        // another variant with the same value (which is the common thing to
+        // do).
+        let mut constified_variants = vec![];
+
+        let mut iter = self.variants().iter().peekable();
+
+        while iter.peek().is_some() || !constified_variants.is_empty() {
+            let variant = iter.next()
+                .unwrap_or_else(|| constified_variants.remove(0));
```

Comment:
You can probably use `VecDeque` instead of `Vec` for `constified_variants`, and change the three lines above to something like
```rust
while let Some(variant) = iter.next().or_else(|| constified_variants.pop_front()) {
    // ...
}
```